x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/MigrateAction.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.xpack.core.ilm.Step.StepKey;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A {@link LifecycleAction} which enables or disables the automatic migration of data between
+ * {@link org.elasticsearch.xpack.core.DataTier}s.
+ */
+public class MigrateAction implements LifecycleAction {
+    public static final String NAME = "migrate";
+    public static final ParseField ENABLED_FIELD = new ParseField("enabled");
+
+    private static final ConstructingObjectParser<MigrateAction, Void> PARSER = new ConstructingObjectParser<>(NAME,
+        a -> new MigrateAction((boolean) a[0]));
+
+    static {
+        PARSER.declareBoolean(ConstructingObjectParser.constructorArg(), ENABLED_FIELD);
+    }
+
+    private final boolean enabled;
+
+    public static MigrateAction parse(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    public MigrateAction(boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    public MigrateAction(StreamInput in) throws IOException {
+        this(in.readBoolean());
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeBoolean(enabled);
+    }
+
+    @Override
+    public String getWriteableName() {
+        return NAME;
+    }
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(ENABLED_FIELD.getPreferredName(), enabled);
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public boolean isSafeAction() {
+        return true;
+    }
+
+    @Override
+    public List<Step> toSteps(Client client, String phase, StepKey nextStepKey) {
+        if (enabled) {
+            Map<String, String> include = Map.of("_tier", "data_" + phase);
+            AllocateAction migrateDataAction = new AllocateAction(null, include, null, null);
```

Comment:
should we remove the possible`_require` and `_exclude` settings (which might've been set manually before) to make sure they don't invalidate the "migrate to the next data tier" goal?

Reply:
That's a good question, I'm not sure we'd want to piecemeal remove all of them, because they could be something for an *additional* attribute that we want to preserve between phases.

I will have to think on it a bit, it also makes me wonder whether we should make it configurable whether all other index-level allocation filtering settings are removed/preserved when doing the migration.